packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1103,6 +1104,46 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
   /// Override in subclasses with children and call the visitor for each child.
   void visitChildren(RenderObjectVisitor visitor) { }
 
+  /// Capture an image of the current state of the render object and its
+  /// children.
+  ///
+  /// The returned [ui.Image] has uncompressed raw bytes in the dimensions of
+  /// the render object, multiplied by the [pixelRatio].
+  ///
+  /// The property [isRepaintBoundary] must be true for [toImage] to work, and
+  /// the render object must not need painting (i.e. [debugNeedsPaint] must be
+  /// false).
+  ///
+  /// The [pixelRatio] describes the scale between the logical pixels and the
+  /// size of the output image. It is independent of the
+  /// [window.devicePixelRatio] for the device, so specifying 1.0 (the default)
+  /// will give you a 1:1 mapping between logical pixels and the output pixels
+  /// in the image.
+  ///
+  /// See also:
+  ///
+  ///  * [ui.Scene.toImage] for more information about the image returned.
+  Future<ui.Image> toImage({double pixelRatio: 1.0}) async {
```

Comment:
To avoid people depending on particular widgets that happen to be repaint boundaries but might not always be, I recommend putting this on RenderRepaintBoundary.

Reply:
Yeah, I guess that makes sense.  It means that they'll have to cast the result when they get the render object in order to call this function, but that's not horrible.